src/Content/Item.php

```diff
@@ -1386,4 +1398,140 @@ public function getPlayerIframe(string $player_url, ?int $width, ?int $height):
 		$return .= '<iframe ' . $attributes . ' width="100%" frameborder="0" allow="fullscreen, picture-in-picture" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>';
 		return $return;
 	}
+
+	/**
+	 * Embed remote media, that had been added with the [embed] element
+	 *
+	 * @param string $html    The already rendered HTML output
+	 * @param integer $uid    The user the output is rendered for
+	 * @param integer $uri_id The uri-id of the item that is rendered
+	 * @return string
+	 */
+	public function addEmbed(string $html, int $uid, int $uri_id): string
+	{
+		if ($html == '') {
+			return $html;
+		}
+
+		$allow_embed = $this->pConfig->get($uid, 'system', 'embed_remote_media', false);
+
+		$changed = false;
+
+		$tmp = new DOMDocument();
+		$doc = new DOMDocument();
+		@$doc->loadHTML(mb_convert_encoding('<span>' . $html . '</span>', 'HTML-ENTITIES', "UTF-8"), LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
+		$xpath = new DOMXPath($doc);
+		$list  = $xpath->query('//a[@class="embed"]');
+		foreach ($list as $node) {
+			$href = '';
+			if ($node->attributes->length) {
+				foreach ($node->attributes as $attribute) {
+					if ($attribute->name == 'href') {
+						$href = $attribute->value;
+						break;
+					}
+				}
+			}
+			if (empty($href)) {
+				continue;
+			}
+
+			if ($uri_id > 0) {
+				$media = $this->postMediaRepository->selectByURL($uri_id, $href, [Post\Media::HTML, Post\Media::AUDIO, Post\Media::VIDEO, Post\Media::HLS]);
+			}
+			if (empty($media)) {
```

Comment:
```suggestion
			if (!isset($media)) {
```

because it's either `PostMedia` or `NULL`

Reply:
I didn't knew, that `isset` on an existing variable with a `null` would return false. Is changed now.